File: src/model/player/player_container.py

class Player_Container:
	def __init__(self,player,collection):
		self.player = player
		self.collection = collection

	def xml_output(self,player_type):
		out_str = ""
		out_str += "<player>"
		out_str += self.player.xml_output(player_type)
		out_str += "</player>"
		out_str += "<collection>"
		out_str += self.collection.xml_output(player_type)
		out_str += "</collection>"
		return out_str

File: src/jhost/model/PlayerContainer.java
package model;

import org.w3c.dom.Element;

import pplib.XmlParser;
import pplib.exceptions.*;

import model.Player;
import model.Deck;

public class PlayerContainer{
	Player player;
	Deck deck;

	public PlayerContainer(Player player, Deck deck){
		this.player = player;
		this.deck = deck;
	}

	public PlayerContainer(XmlParser xmlParser, Element element) throws PPXmlException {
		this.player = new Player(xmlParser,xmlParser.parseElement(element,"player"));
		this.deck = new Deck(xmlParser,xmlParser.parseElement(element,"deck"));
	}

	public String xmlOutput(int playerType) throws PPGameActionException{
		String xml = "<player>"+this.player.xmlOutput(playerType)+"</player>";
		xml += "<collection>"+this.deck.xmlOutput(playerType)+"</collection>";
		return xml;
	}

	public Player getPlayer(){
		return this.player;
	}

	public Deck getDeck(){
		return this.deck;
	}
}
